extensions/virtual-threads/deployment/src/main/java/io/quarkus/virtual/threads/VirtualThreadsProcessor.java

```diff
@@ -12,8 +21,20 @@ public class VirtualThreadsProcessor {
     @Record(ExecutionTime.STATIC_INIT)
     public void setup(VirtualThreadsConfig config, VirtualThreadsRecorder recorder,
             ShutdownContextBuildItem shutdownContextBuildItem,
-            LaunchModeBuildItem launchModeBuildItem) {
+            LaunchModeBuildItem launchModeBuildItem,
+            BuildProducer<AdditionalBeanBuildItem> beans,
+            BuildProducer<SyntheticBeanBuildItem> producer) {
+        beans.produce(new AdditionalBeanBuildItem(VirtualThreads.class));
         recorder.setupVirtualThreads(config, shutdownContextBuildItem, launchModeBuildItem.getLaunchMode());
+        producer.produce(
+                SyntheticBeanBuildItem.configure(ExecutorService.class)
+                        .addType(Executor.class)
+                        .addQualifier(AnnotationInstance.builder(VirtualThreads.class).build())
+                        .scope(BuiltinScope.APPLICATION.getInfo())
+                        .unremovable()
```

Comment:
I wonder if it has to be unremovable? I mean do we expect users to do `CDI.current().select(ExecutorService.class, VirtualThreads.Literal.INSTANCE)`? In theory, if you really need to access it programmatically you can simply do `VirtualThreadsRecorder.getCurrent()` :shrug:

Reply:
Yes, I don't believe it should be unremovable. Good catch @mkouba!

Reply:
Good point! I'll remove it.